### DISCUSSION TITLE
工单一直在确认中，实际语句已执行完毕

### DISCUSSION THREAD

**Discussion by bluerfish** [hhyo/Archery#1924] [General]
手动部署1.9.1版本
工单的语句比较多，有4万多条update,在执行期间，一直在确认中。从Django-后台看到[sqlreview-execute] 在执行400多秒的时候已成功结束，而且业务也确认执行成功。在 qcluster的日志中看到，return hook sql.utils.execute_sql.execute_callback failed on [sqlreview-execute-23] because (2013, 'Lost connection to MySQL server during query')。MAX_EXECUTION_TIME 我设置的是120秒。后续我把状态手工修改了。我想问下，这个由于MAX_EXECUTION_TIME 导致的吗，这个不是查询的超时时间吗。另外，我们这边服务器的配置资源特别高（目前是64C256G）。我们这边想在审核和执行过程中更快速一些，有哪些可以优化的呢。或者说要修改哪个地方会多占用资源呢。

**Comment by hhyo** [comment]
这种情况一般是执行的结果集过大导致更新失败，可以调整Archery数据库的max_allowed_packet值

关于执行效率，平台在执行时做了一些限速处理，每执行100行会sleep200ms，有需要可以自行调整下代码https://github.com/hhyo/Archery/blob/029ebab887e18642f056fb562fd309a9776788c0/sql/engines/goinception.py#L111

**Reply by bluerfish** [reply]
我把--sleep=200;--sleep_rows=100 这两个参数去掉了。重启后，新执行工单（DDL加索引--256张表）的时候，sql.utils.execute_sql.execute中的时间是 945.936074秒。在工单审核界面中。所有语句的执行耗时相加 是521秒。在执行工单的前5分钟（确认中状态），数据库中是没用进行alter操作的。这个是在检测数据字典对吧，请问下这个可以在哪里优化呢。另外，在执行中的时候，当前工单界面中：已经执行完的语句的“当前阶段”是Audit Completed，后续是所有工单执行完成后，所有的语句的当前阶段都显示“Execute Successfully”
